### DISCUSSION TITLE
posts/2023/05/better-code-image-processing-hugo-render-hook-edition/

### DISCUSSION THREAD

**Discussion by giscus** [brycewray/comments#42] [Announcements]
# posts/2023/05/better-code-image-processing-hugo-render-hook-edition/

If you’re a fan of the standard Markdown syntax for including an image, this post is for you.

https://www.brycewray.com/posts/2023/05/better-code-image-processing-hugo-render-hook-edition/

**Comment by boetiusj** [comment]
Hi, 
This is an elegant bit of code, just what I've been hunting for. 
I get an error: nil pointer evaluating resource.Resourse.Resize
{{- $actualImg := $src.Resize (print "640x jpg " $filter) -}}

Any suggestions? Thanks, Mike

**Reply by brycewray** [reply]
@boetiusj Just a guess, but it sounds as if it’s not finding the image where it expects it to be. 

See also:

https://discourse.gohugo.io/t/nil-pointer-evaluating-resource-resource-resize/23236

**Reply by boetiusj** [reply]
THANKS for your speedy response. I'll dig into it while I dodge Christmas travel.
